Content.Server/Atmos/EntitySystems/AtmosphereSystem.HighPressureDelta.cs

```diff
@@ -80,7 +85,19 @@ private void AddMobMovedByPressure(EntityUid uid, MovedByPressureComponent compo
 
             foreach (var (id, fixture) in fixtures.Fixtures)
             {
-                _physics.RemoveCollisionMask(uid, id, fixture, (int) CollisionGroup.TableLayer, manager: fixtures);
+                // Mark fixtures that has TableLayer removed
+                if ((fixture.CollisionMask & (int)CollisionGroup.TableLayer) != 0)
+                    {
+                        component.TableLayerRemoved[id] = true;
+                        _physics.RemoveCollisionMask(uid, id, fixture, (int)CollisionGroup.TableLayer, manager: fixtures);
+                    }
+                else
+                    {
+                        if (!component.TableLayerRemoved.ContainsKey(id))
+                        {
+                            component.TableLayerRemoved[id] = false;
```

Comment:
Why store a hash map of layers and removed/not removed, instead of storing a hash set of only layers removed?

Reply:
haven't even thought about it lol